graphql/resolvers.go

```diff
@@ -3266,10 +3252,23 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 		}
 	}
 
-	for _, v := range versions {
-		if activatedVersionCount == limit {
+	index := 0
+	versionsCheckedCount := 0
+	for matchingVersionCount < limit {
```

Comment:
Can we add a comment here to describe the loop, i.e. something about how we're looping through each version returned from GetMainlineCommitVersionsWithOptions and will call the query again if we run out? We don't use while loops often so i think it'll help us for the future